util/resourcegrouptag/resource_group_tag.go

```diff
@@ -0,0 +1,73 @@
+package resourcegrouptag
+
+import (
+	"encoding/hex"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+const (
+	resourceGroupTagPrefixSQLDigest = byte(1)
+)
+
+// EncodeResourceGroupTag encodes sqlDigest into resource group tag.
+// A resource group tag can be carried in the Context field of TiKV requests, which is a byte array, and sent to TiKV as
+// diagnostic information. Currently it contains only the SQL Digest, and the codec method is naive but extendable.
+// This function doesn't return error. When there's some error, which can only be caused by unexpected format of the
+// arguments, it simply returns an empty result.
+func EncodeResourceGroupTag(sqlDigest string) []byte {
+	if len(sqlDigest) == 0 {
+		return nil
+	}
+	if len(sqlDigest) >= 512 {
+		logutil.BgLogger().Warn("failed to encode sql digest to resource group tag: length too long", zap.String("sqlDigest", sqlDigest))
+		return nil
+	}
+
+	res := make([]byte, 3+len(sqlDigest)/2)
+	const encodingVersion = 1
+	res[0] = encodingVersion
+	res[1] = resourceGroupTagPrefixSQLDigest
+	res[2] = byte(len(sqlDigest) / 2)
+	_, err := hex.Decode(res[3:], []byte(sqlDigest))
```

Comment:
Would you please add some comment about the current encoding methods, eg. why we do `hex.Decode` here instead of just pass `sqlDigest` though.